src/main/java/ladder/domain/LadderGame.java

```diff
@@ -0,0 +1,43 @@
+package ladder.domain;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class LadderGame {
+    private final List<PlayerName> players;
+    private final Ladder ladder;
+
+    public LadderGame(List<PlayerName> players, Ladder ladder) {
+        this.players = players;
+        this.ladder = ladder;
+    }
+
+    public List<String> playGame() {
+        List<PlayerName> playerNames = new ArrayList<>(players);
+        for (int i = 0; i < ladder.getHeight(); i++) {
+            playerNames = playOneLine(playerNames, i);
+        }
+        return getResult(playerNames);
+    }
+
+    private List<PlayerName> playOneLine(List<PlayerName> playerNames, int height) {
+        List<Line> lines = ladder.getLines();
+        Line line = lines.get(height);
+
+        Line line1 = line.addGap();
+        for (int i = 0; i < line.getSticks().size(); i++) {
+            int movePosition = line1.move(i);
+            Collections.swap(playerNames, i, movePosition);
+        }
+        return playerNames;
+    }
+
+    private static List<String> getResult(List<PlayerName> players) {
```

Comment:
```suggestion
    private List<String> getResult(List<PlayerName> players) {
```
메서드 분리할 때 static 이 포함되죠? 😂
이거 method 분리 시도했을 때 옆에 설정 버튼 누르고 체크박스 제거하면 default 로 static 안 붙게 할 수 있어요 :

<img width="340" alt="Screenshot 2024-03-02 at 4 04 10 PM" src="https://github.com/woowacourse/java-ladder/assets/48251668/c8092316-29be-4abb-8006-f36ed7380225">